src/proj_types.jl

```diff
@@ -0,0 +1,50 @@
+## Types
+
+# Projection context.  TODO: Will this be exposed?
+#type Context
+#    rep::Ptr{Void} # Pointer to internal projCtx struct
+#end
+
+@doc """
+Cartographic projection type
+""" ->
+type Projection
+    #ctx::Context   # Projection context object
+    rep::Ptr{Void} # Pointer to internal projPJ struct
+    
+    # [geod]: a structure containing the parameters of the spheroid
+    # some of the fields in [geod] are mildly duplicative of the information
+    # available in [rep], which can be exposed only through _get_spheroid_defn
+
+    # It's needed as an argument for computing great circle distances though,
+    # and for most applications, users will only have to deal with a small number
+    # of projection objects, and so we precompute it for each Projection
+    geod::geod_geodesic
```

Comment:
> Let's not do this in the current PR, it's already pretty huge.

You mean: not define
```julia
type Projection
    ctx::ProjCtx
    proj::Proj
    geod::GeodCtx
end
```
in this PR? I can see what you're saying, but (to my knowledge) it isn't common practice in julia wrapper libraries. I would write
```julia
type Projection
    ptr::Ptr{Void}
    geod::geod_geodesic
end
```
(e.g. https://github.com/JuliaGeo/LibGEOS.jl/blob/master/src/geos_types.jl) to make the correspondence to pointers obvious from the attribute name, but wouldn't go so far as to wrap it in additional types.

IMO, writing `proj.ptr` provides the same level of warning as writing `proj.rep.ctx`? And if a user is aware of the existence of the `proj.ptr` attribute, and intends to use it, that'll be undocumented behavior (and hence their responsibility)